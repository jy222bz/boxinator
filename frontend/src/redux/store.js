import { combineReducers, createStore } from "@reduxjs/toolkit";
import sliceReducer from "./reduceSlice";

/**
 *  Making a store for the reducer.
 * 
 * @author Jacob Yousif
 * 
 *  @type {*} 
 * 
 */
const reducer = combineReducers({
 value: sliceReducer
});


const store = createStore(reducer);

export default store;