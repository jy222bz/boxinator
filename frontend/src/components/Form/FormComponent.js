import React from "react"
import isValidInput from '../../models/input-validator';
import { ToastContainer, toast } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import Color from '../color/ColorComponent'
import '../../css/master.css';
import Service from '../../services/rest-service'
import { connect } from "react-redux";
import { setNewState } from "../../redux/reduceSlice";
import hexRgb from 'hex-rgb';


/**
 * A Form component for the add box operation.
 * @author Jacob Yousif
 *
 * @class FormComponent
 * @extends {React.Component}
 */

class FormComponent extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            name: '',
            color: '',
            weight: 0,
            country: ''
        }
        this.ref = React.createRef()

    }

    setName = val => {
        this.setState({ name: val })
    }

    setWeight = val => {
        this.setState({ weight: val })
    }

    setCountry = val => {
        this.setState({ country: val })
    }

    show = () => {
        const div = document.getElementById('color');
        if (div.style.display === 'none') {
            div.style.display = 'block'
        } else {
            div.style.display = 'none'
        }
    }


    onInputChage = value => {
        this.setState({
            color: value.hex
        })
    }

    onSubmit = (e) => {
        e.preventDefault();
        try {
            isValidInput(this.state.name, this.state.country, this.state.color, this.state.weight)
            let data = { name: this.state.name.toUpperCase(), weight: this.state.weight, color: hexRgb(this.state.color, { format: "css" }), country: this.state.country.toUpperCase() }

            Service.post(data)
            toast.success('The information was successfully inserted into the database.', { hideProgressBar: true });
            this.setState({ country: '', name: '', color: '', weight: 0 })
            this.ref.current.init()
            this.props.setNewState(true);
            const div = document.getElementById('color');
            div.style.display = 'none'
        } catch (err) {
            toast.error(err.message, { hideProgressBar: true });
        }
    }

    render() {
        return (
            <section data-testid='id__section_main' className="sec2" onSubmit={this.onSubmit}>
                <div data-testid='id__div' >
                    <h2 data-testid='id__headline'>The Required Info</h2>
                    <form data-testid='id__form'>
                        <label data-testid='id__label__name'>Name:</label><br />
                        <input data-testid='id__input__name' type="text" id="name" name="name" placeholder="name" value={this.state.name} onChange={(e) => this.setName(e.target.value)}></input><br />
                        <label data-testid='id__label__weight'>Weight:</label><br />
                        <input data-testid='id__input__weight' type="number" id="number" name="weight" placeholder="weight in KG" value={this.state.weight} onChange={(e) => this.setWeight(e.target.value)}></input><br />
                        <input readOnly="readonly" className="label1" type="text" data-testid='id__label__color' id='color__label' onClick={this.show} defaultValue="Click here to Close/Show the Color Picker"></input><br />
                        <div data-testid='id__div__color' className="color" id="color">
                            <Color ref={this.ref} onSubmit={this.init} color={this.state.color} onChange={this.onInputChage} />
                        </div ><br />
                        <div data-testid='id__div__select'  >
                            <label data-testid='id__label__country'   >Country:</label><br></br>
                            <select data-testid='id__select__country' className="select" id="slct" value={this.state.country} onChange={(e) => this.setCountry(e.target.value)}>
                                <option data-testid='id__select__1' value="">Select a country</option>
                                <option data-testid='id__select__2' value="Sweden">Sweden</option>
                                <option data-testid='id__select__3' value="China">China</option>
                                <option data-testid='id__select__4' value="Brazil">Brazil</option>
                                <option data-testid='id__select__5' value="Australia">Australia</option>
                            </select>
                        </div><br />
                        <input data-testid='id__input__submit' type="submit" value="Save"></input> <br />
                    </form>
                </div>
                <ToastContainer />
            </section>
        )
    }
}

/**
 * A function to map the the state to the props of the object.
 *
 * @param {*} state
 */
const mapStateToProps = (state) => ({
    update: state.isNewState
})

export default connect(mapStateToProps, { setNewState })(FormComponent);

