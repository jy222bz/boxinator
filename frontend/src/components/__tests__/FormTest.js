import React from 'react'
import ReactDOM from 'react-dom'
import FormComponent from '../Form/FormComponent'
import { render, screen, cleanup } from '@testing-library/react'
import { Provider } from 'react-redux';
import store from '../../redux/store'
import renderer from 'react-test-renderer'

/**
 * A Teste Suit.
 * @author Jacob Yousif
 *
 */

afterEach(() => {
 cleanup()
})


it('Renders without any faults.', () => {
 const sut = document.createElement('section')
 ReactDOM.render(<Provider store={store}>
  <FormComponent />
 </Provider>, sut)
})

test('It should have a main section.', () => {
 render(<Provider store={store}>
  <FormComponent />
 </Provider>);
 const sut = screen.getByTestId('id__section_main')
 expect(sut).toBeInTheDocument();

});

test('It should have a div in the section.', () => {
 render(<Provider store={store}>
  <FormComponent />
 </Provider>);
 const sut = screen.getByTestId('id__div')
 expect(sut).toBeInTheDocument();

});

test('It should have a headline and text in it.', () => {
 render(<Provider store={store}>
  <FormComponent />
 </Provider>);
 const sut = screen.getByTestId('id__headline');
 expect(sut).toBeInTheDocument();
 expect(sut).toHaveTextContent('The Required Info')
});


test('It should have a form', () => {
 render(<Provider store={store}>
  <FormComponent />
 </Provider>);
 const sut = screen.getByTestId('id__form')
 expect(sut).toBeInTheDocument();
});

test('It should have a label for the name.', () => {
 render(<Provider store={store}>
  <FormComponent />
 </Provider>);
 const sut = screen.getByTestId('id__label__name')
 expect(sut).toBeInTheDocument();
});


test('It should have a label for the weight.', () => {
 render(<Provider store={store}>
  <FormComponent />
 </Provider>);
 const sut = screen.getByTestId('id__label__weight')
 expect(sut).toBeInTheDocument();
});

test('It should have a label for the color.', () => {
 render(<Provider store={store}>
  <FormComponent />
 </Provider>);
 const sut = screen.getByTestId('id__label__color')
 expect(sut).toBeInTheDocument();
});

test('It should have a label for the country.', () => {
 render(<Provider store={store}>
  <FormComponent />
 </Provider>);
 const sut = screen.getByTestId('id__label__country')
 expect(sut).toBeInTheDocument();
});

test('It should have a div for the select.', () => {
 render(<Provider store={store}>
  <FormComponent />
 </Provider>);
 const sut = screen.getByTestId('id__div__select')
 expect(sut).toBeInTheDocument();
});

test('It should have an input for the name.', () => {
 render(<Provider store={store}>
  <FormComponent />
 </Provider>);
 const sut = screen.getByTestId('id__input__name')
 expect(sut).toBeInTheDocument();
});

test('It should have an input for the weight.', () => {
 render(<Provider store={store}>
  <FormComponent />
 </Provider>);
 const sut = screen.getByTestId('id__input__weight')
 expect(sut).toBeInTheDocument();
});

test('It should have a select.', () => {
 render(<Provider store={store}>
  <FormComponent />
 </Provider>);
 const sut = screen.getByTestId('id__select__country')
 expect(sut).toBeInTheDocument();
});

test('It should have a select option for option 1.', () => {
 render(<Provider store={store}>
  <FormComponent />
 </Provider>);
 const sut = screen.getByTestId('id__select__1')
 expect(sut).toBeInTheDocument();
 expect(sut).toHaveTextContent('Select a country')
});


test('It should have a select option for option 2.', () => {
 render(<Provider store={store}>
  <FormComponent />
 </Provider>);
 const sut = screen.getByTestId('id__select__2')
 expect(sut).toBeInTheDocument();
 expect(sut).toHaveTextContent('Sweden')
});

test('It should have a select option for option 3.', () => {
 render(<Provider store={store}>
  <FormComponent />
 </Provider>);
 const sut = screen.getByTestId('id__select__3')
 expect(sut).toBeInTheDocument();
 expect(sut).toHaveTextContent('China')
});

test('It should have a select option for option 4.', () => {
 render(<Provider store={store}>
  <FormComponent />
 </Provider>);
 const sut = screen.getByTestId('id__select__4')
 expect(sut).toBeInTheDocument();
 expect(sut).toHaveTextContent('Brazil')
});

test('It should have a select option for option 5.', () => {
 render(<Provider store={store}>
  <FormComponent />
 </Provider>);
 const sut = screen.getByTestId('id__select__5')
 expect(sut).toBeInTheDocument();
 expect(sut).toHaveTextContent('Australia')
});

test('It should have an input for submit.', () => {
 render(<Provider store={store}>
  <FormComponent />
 </Provider>);
 const sut = screen.getByTestId('id__input__submit')
 expect(sut).toBeInTheDocument();
});


test('It should have a placeholder for the name input.', () => {
 render(<Provider store={store}>
  <FormComponent />
 </Provider>);
 const sut = screen.queryByPlaceholderText(/name/i)
 expect(sut).toBeInTheDocument();
});

test('It should have a placeholder for the weight input.', () => {
 render(<Provider store={store}>
  <FormComponent />
 </Provider>);
 const sut = screen.queryByPlaceholderText(/weight/i)
 expect(sut).toBeInTheDocument();
});

test('It should match snapshot', () => {
 const sut = renderer.create(<Provider store={store}>
  <FormComponent />
 </Provider>).toJSON()
 expect(sut).toMatchSnapshot()
});


