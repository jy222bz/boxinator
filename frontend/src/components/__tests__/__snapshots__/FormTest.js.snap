// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should match snapshot 1`] = `
<section
  className="sec2"
  data-testid="id__section_main"
  onSubmit={[Function]}
>
  <div
    data-testid="id__div"
  >
    <h2
      data-testid="id__headline"
    >
      The Required Info
    </h2>
    <form
      data-testid="id__form"
    >
      <label
        data-testid="id__label__name"
      >
        Name:
      </label>
      <br />
      <input
        data-testid="id__input__name"
        id="name"
        name="name"
        onChange={[Function]}
        placeholder="name"
        type="text"
        value=""
      />
      <br />
      <label
        data-testid="id__label__weight"
      >
        Weight:
      </label>
      <br />
      <input
        data-testid="id__input__weight"
        id="number"
        name="weight"
        onChange={[Function]}
        placeholder="weight in KG"
        type="number"
        value={0}
      />
      <br />
      <input
        className="label1"
        data-testid="id__label__color"
        defaultValue="Click here to Close/Show the Color Picker"
        id="color__label"
        onClick={[Function]}
        readOnly="readonly"
        type="text"
      />
      <br />
      <div
        className="color"
        data-testid="id__div__color"
        id="color"
      >
        <div
          className="github-picker "
          style={
            Object {
              "MozBorderRadius": "4px",
              "MozBoxShadow": "0 3px 12px rgba(0,0,0,0.15)",
              "OBorderRadius": "4px",
              "OBoxShadow": "0 3px 12px rgba(0,0,0,0.15)",
              "WebkitBorderRadius": "4px",
              "WebkitBoxShadow": "0 3px 12px rgba(0,0,0,0.15)",
              "background": "#fff",
              "border": "1px solid rgba(0,0,0,0.2)",
              "borderRadius": "4px",
              "boxShadow": "0 3px 12px rgba(0,0,0,0.15)",
              "display": "flex",
              "flexWrap": "wrap",
              "msBorderRadius": "4px",
              "msBoxShadow": "0 3px 12px rgba(0,0,0,0.15)",
              "padding": "5px",
              "position": "relative",
              "width": "60%",
            }
          }
        >
          <div
            style={
              Object {
                "border": "8px solid transparent",
                "borderBottomColor": "rgba(0,0,0,0.15)",
                "left": "9px",
                "position": "absolute",
                "top": "-16px",
              }
            }
          />
          <div
            style={
              Object {
                "border": "7px solid transparent",
                "borderBottomColor": "#fff",
                "left": "10px",
                "position": "absolute",
                "top": "-14px",
              }
            }
          />
          <span
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <div
              style={
                Object {
                  "fontSize": "0",
                  "height": "25px",
                  "width": "25px",
                }
              }
            >
              <span
                onBlur={[Function]}
                onFocus={[Function]}
              >
                <div
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  style={
                    Object {
                      "background": "#FFB74D",
                      "cursor": "pointer",
                      "height": "100%",
                      "outline": "none",
                      "position": "relative",
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  title="#FFB74D"
                />
              </span>
            </div>
          </span>
          <span
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <div
              style={
                Object {
                  "fontSize": "0",
                  "height": "25px",
                  "width": "25px",
                }
              }
            >
              <span
                onBlur={[Function]}
                onFocus={[Function]}
              >
                <div
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  style={
                    Object {
                      "background": "#008B02",
                      "cursor": "pointer",
                      "height": "100%",
                      "outline": "none",
                      "position": "relative",
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  title="#008B02"
                />
              </span>
            </div>
          </span>
          <span
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <div
              style={
                Object {
                  "fontSize": "0",
                  "height": "25px",
                  "width": "25px",
                }
              }
            >
              <span
                onBlur={[Function]}
                onFocus={[Function]}
              >
                <div
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  style={
                    Object {
                      "background": "#EB9694",
                      "cursor": "pointer",
                      "height": "100%",
                      "outline": "none",
                      "position": "relative",
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  title="#EB9694"
                />
              </span>
            </div>
          </span>
          <span
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <div
              style={
                Object {
                  "fontSize": "0",
                  "height": "25px",
                  "width": "25px",
                }
              }
            >
              <span
                onBlur={[Function]}
                onFocus={[Function]}
              >
                <div
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  style={
                    Object {
                      "background": "#FAD0C3",
                      "cursor": "pointer",
                      "height": "100%",
                      "outline": "none",
                      "position": "relative",
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  title="#FAD0C3"
                />
              </span>
            </div>
          </span>
          <span
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <div
              style={
                Object {
                  "fontSize": "0",
                  "height": "25px",
                  "width": "25px",
                }
              }
            >
              <span
                onBlur={[Function]}
                onFocus={[Function]}
              >
                <div
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  style={
                    Object {
                      "background": "#FEF3BD",
                      "cursor": "pointer",
                      "height": "100%",
                      "outline": "none",
                      "position": "relative",
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  title="#FEF3BD"
                />
              </span>
            </div>
          </span>
          <span
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <div
              style={
                Object {
                  "fontSize": "0",
                  "height": "25px",
                  "width": "25px",
                }
              }
            >
              <span
                onBlur={[Function]}
                onFocus={[Function]}
              >
                <div
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  style={
                    Object {
                      "background": "#e84393",
                      "cursor": "pointer",
                      "height": "100%",
                      "outline": "none",
                      "position": "relative",
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  title="#e84393"
                />
              </span>
            </div>
          </span>
          <span
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <div
              style={
                Object {
                  "fontSize": "0",
                  "height": "25px",
                  "width": "25px",
                }
              }
            >
              <span
                onBlur={[Function]}
                onFocus={[Function]}
              >
                <div
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  style={
                    Object {
                      "background": "#2d3436",
                      "cursor": "pointer",
                      "height": "100%",
                      "outline": "none",
                      "position": "relative",
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  title="#2d3436"
                />
              </span>
            </div>
          </span>
          <span
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <div
              style={
                Object {
                  "fontSize": "0",
                  "height": "25px",
                  "width": "25px",
                }
              }
            >
              <span
                onBlur={[Function]}
                onFocus={[Function]}
              >
                <div
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  style={
                    Object {
                      "background": "#C1E1C5",
                      "cursor": "pointer",
                      "height": "100%",
                      "outline": "none",
                      "position": "relative",
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  title="#C1E1C5"
                />
              </span>
            </div>
          </span>
          <span
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <div
              style={
                Object {
                  "fontSize": "0",
                  "height": "25px",
                  "width": "25px",
                }
              }
            >
              <span
                onBlur={[Function]}
                onFocus={[Function]}
              >
                <div
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  style={
                    Object {
                      "background": "#00b894",
                      "cursor": "pointer",
                      "height": "100%",
                      "outline": "none",
                      "position": "relative",
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  title="#00b894"
                />
              </span>
            </div>
          </span>
          <span
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <div
              style={
                Object {
                  "fontSize": "0",
                  "height": "25px",
                  "width": "25px",
                }
              }
            >
              <span
                onBlur={[Function]}
                onFocus={[Function]}
              >
                <div
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  style={
                    Object {
                      "background": "#7f8c8d",
                      "cursor": "pointer",
                      "height": "100%",
                      "outline": "none",
                      "position": "relative",
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  title="#7f8c8d"
                />
              </span>
            </div>
          </span>
          <span
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <div
              style={
                Object {
                  "fontSize": "0",
                  "height": "25px",
                  "width": "25px",
                }
              }
            >
              <span
                onBlur={[Function]}
                onFocus={[Function]}
              >
                <div
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  style={
                    Object {
                      "background": "#fff",
                      "cursor": "pointer",
                      "height": "100%",
                      "outline": "none",
                      "position": "relative",
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  title="#fff"
                />
              </span>
            </div>
          </span>
          <span
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <div
              style={
                Object {
                  "fontSize": "0",
                  "height": "25px",
                  "width": "25px",
                }
              }
            >
              <span
                onBlur={[Function]}
                onFocus={[Function]}
              >
                <div
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  style={
                    Object {
                      "background": "#006266",
                      "cursor": "pointer",
                      "height": "100%",
                      "outline": "none",
                      "position": "relative",
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  title="#006266"
                />
              </span>
            </div>
          </span>
          <span
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <div
              style={
                Object {
                  "fontSize": "0",
                  "height": "25px",
                  "width": "25px",
                }
              }
            >
              <span
                onBlur={[Function]}
                onFocus={[Function]}
              >
                <div
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  style={
                    Object {
                      "background": "#EA2027",
                      "cursor": "pointer",
                      "height": "100%",
                      "outline": "none",
                      "position": "relative",
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  title="#EA2027"
                />
              </span>
            </div>
          </span>
          <span
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <div
              style={
                Object {
                  "fontSize": "0",
                  "height": "25px",
                  "width": "25px",
                }
              }
            >
              <span
                onBlur={[Function]}
                onFocus={[Function]}
              >
                <div
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  style={
                    Object {
                      "background": "#795548",
                      "cursor": "pointer",
                      "height": "100%",
                      "outline": "none",
                      "position": "relative",
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  title="#795548"
                />
              </span>
            </div>
          </span>
          <span
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <div
              style={
                Object {
                  "fontSize": "0",
                  "height": "25px",
                  "width": "25px",
                }
              }
            >
              <span
                onBlur={[Function]}
                onFocus={[Function]}
              >
                <div
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  style={
                    Object {
                      "background": "#f44336",
                      "cursor": "pointer",
                      "height": "100%",
                      "outline": "none",
                      "position": "relative",
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  title="#f44336"
                />
              </span>
            </div>
          </span>
          <span
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <div
              style={
                Object {
                  "fontSize": "0",
                  "height": "25px",
                  "width": "25px",
                }
              }
            >
              <span
                onBlur={[Function]}
                onFocus={[Function]}
              >
                <div
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  style={
                    Object {
                      "background": "#e91e63",
                      "cursor": "pointer",
                      "height": "100%",
                      "outline": "none",
                      "position": "relative",
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  title="#e91e63"
                />
              </span>
            </div>
          </span>
        </div>
      </div>
      <br />
      <div
        data-testid="id__div__select"
      >
        <label
          data-testid="id__label__country"
        >
          Country:
        </label>
        <br />
        <select
          className="select"
          data-testid="id__select__country"
          id="slct"
          onChange={[Function]}
          value=""
        >
          <option
            data-testid="id__select__1"
            value=""
          >
            Select a country
          </option>
          <option
            data-testid="id__select__2"
            value="Sweden"
          >
            Sweden
          </option>
          <option
            data-testid="id__select__3"
            value="China"
          >
            China
          </option>
          <option
            data-testid="id__select__4"
            value="Brazil"
          >
            Brazil
          </option>
          <option
            data-testid="id__select__5"
            value="Australia"
          >
            Australia
          </option>
        </select>
      </div>
      <br />
      <input
        data-testid="id__input__submit"
        type="submit"
        value="Save"
      />
       
      <br />
    </form>
  </div>
  <div
    className="Toastify"
  />
</section>
`;
